code.js

```diff
@@ -1,3 +1,63 @@
 function tsp_ls(distance_matrix) {
-    return -1;
+    //From this comment to the next creates list that is a permutation of the nodes
+    let nodes = [];
+    let route = [];
+    for(let i = 0; i < distance_matrix.length; i++)
+        nodes[i] = i;
+    for(let i = distance_matrix.length; i > 0; i--)
+        route[i - 1] = nodes.splice(getRandomInt(i), 1);
+    //I'm "The next"!
+
+    let routeLen = routeCost(distance_matrix, route);
+    let min = Infinity;
+    for(let b = 0; b < 3; b++){ //Because the inner for loop takes a random number, this for loop gives it 3 chances to pick a good random index.
+        for(let i = getRandomInt(distance_matrix.length - 1), k = i + 1, v = 0; i >= 0 && k < distance_matrix.length; v++){
+            // i is selected randomly and k is chosen to be just above it.
+            tempRoute = optSwap(route, i, k);
+            let newRouteLen = routeCost(distance_matrix, tempRoute);
+            if(routeLen > newRouteLen) {
+                routeLen = newRouteLen;
+                route = JSON.parse(JSON.stringify(tempRoute));
```

Comment:
I changed it by removing the "min" variable because it was not necessary, so now routeLen is used at the end. Is there another way I can improve this?